### DISCUSSION TITLE
extending unjsPreset

### DISCUSSION THREAD

**Discussion by amery** [unjs/eslint-config#16] [General]
Hi, I'm trying to build my [own eslint-config](https://github.com/poupe-ui/eslint-config) package on top of `eslint-config-unjs` initially by adding `vue/recommended` and `stylistic/recommended`, but I haven't figured out 4 things, and I would really appreciate any suggestions.
1) how to procedurally extend `RuleOptions` with definitions from the plugins I'm loading
2) how to handle duplicates in `plugins` declarations to be able to work `withNuxt()` and without
3) `ignores: [ '**/dist' ]` doesn't do anything unless it's passed via the `MainConfig`
4) are there types for `eslint-plugin-vue`?

https://github.com/poupe-ui/eslint-config/blob/main/src/config.ts

**Comment by amery** [comment]
to make things weirder `'@stylistic/semi': ['error', 'always'],` only works on `userConfigs` and not on `MainConfig` when stylistic's recommended rules are also loaded. I thought `MainConfig.rules` was the ultimate override.

**Comment by pi0** [comment]
Hi. Quick answer, you need to use something like `[...unjs({}), <your rules>]` or `unjs({}, ...your rules)` the later ones overrides defaults. `mainConfig` overrides unjs rules only for convenience of usage across repos.

**Reply by amery** [reply]
Thank you for clarifying the intended use of mainConfig, behaviour is less confusing now

**Comment by amery** [comment]
```
Argument of type 'ConfigWithExtends' is not assignable to parameter of type 'TypedFlatConfig'.
  Types of property 'ignores' are incompatible.
    Type 'FileSpec[] | undefined' is not assignable to type 'string[] | undefined'.
      Type 'FileSpec[]' is not assignable to type 'string[]'.
        Type 'FileSpec' is not assignable to type 'string'.
          Type '(filePath: string) => boolean' is not assignable to type 'string'.ts(2345)
(parameter) userConfigs: ConfigWithExtends[]
```
^--- this might be the cause of the problem adding ignores

but if I use `typescript-eslint`'s config() to type the rules instead of unjs() `unbuild` explodes :sob:

**Comment by amery** [comment]
https://github.com/poupe-ui/eslint-config/blob/pr-amery-next/src/config.ts <--- this branch illustrates how `unbuild` explodes when using unjsPreset and typescript-eslint's config() side by side. `eslint` using the files via jiti's stub works though.

it's a single commit from the version using only unjsPreset(), without rule types for options added from plugins and ignore problems, and one using tseslint's config() to type my rules next to unjsPreset(), which works for `tsc --noEmit`,  `eslint` and `vscode` over `unbuild --stub` but exploding on `unbuild`.
